.github/workflows/build-test.yml

```diff
@@ -0,0 +1,92 @@
+name: Validate and test
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+    tags-ignore: [ v.* ]
+
+jobs:
+  check-code-style:
+    name: Checks
+    runs-on: ubuntu-22.04
+    if: github.repository == 'akka/akka-diagnostics'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3.1.0
+        with:
+          # we don't know what commit the last tag was it's safer to get entire repo so previousStableVersion resolves
+          fetch-depth: 0
+
+      - name: Checkout GitHub merge
+        if: github.event.pull_request
+        run: |-
+          git fetch origin pull/${{ github.event.pull_request.number }}/merge:scratch
+          git checkout scratch
+      - name: Cache Coursier cache
+        uses: coursier/cache-action@v6.4.0
+
+      - name: Set up JDK 17
+        uses: coursier/setup-action@v1.3.0
+        with:
+          jvm: temurin:1.17.0
+
+      - name: Code style check and binary-compatibility check
+        run: |-
+          cp .jvmopts-ci .jvmopts
+          sbt scalafmtCheckAll scalafmtSbtCheck headerCheck
+  compile-with-scala-2_12:
+    name: Compile with Scala 2.12
+    runs-on: ubuntu-22.04
+    if: github.repository == 'akka/akka-diagnostics'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3.1.0
+        with:
+          # we don't know what commit the last tag was it's safer to get entire repo so previousStableVersion resolves
+          fetch-depth: 0
+      - name: Checkout GitHub merge
+        if: github.event.pull_request
+        run: |-
+          git fetch origin pull/${{ github.event.pull_request.number }}/merge:scratch
+          git checkout scratch
+      - name: Cache Coursier cache
+        uses: coursier/cache-action@v6.4.0
+
+      - name: Set up JDK 11
+        uses: coursier/setup-action@v1.3.0
+        with:
+          jvm: temurin:1.11.0
+
+      - name: Code style check and binary-compatibility check
+        run: |-
+          cp .jvmopts-ci .jvmopts
+          sbt ++2.12 compile Test/compile
+ 
+    name: Run tests
+    runs-on: ubuntu-22.04
+    if: github.repository == 'akka/akka-diagnostics'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3.1.0
+        with:
+          fetch-depth: 0
+
+      - name: Checkout GitHub merge
+        if: github.event.pull_request
+        run: |-
+          git fetch origin pull/${{ github.event.pull_request.number }}/merge:scratch
+          git checkout scratch
+      - name: Cache Coursier cache
+        uses: coursier/cache-action@v6.4.0
+
+      - name: Set up JDK 11
+        uses: coursier/setup-action@v1.3.0
+        with:
+          jvm: temurin:1.11.0
+
+      - name: sbt test
+        run: |-
+          cp .jvmopts-ci .jvmopts
```

Comment:
there is no `.jvmopts-ci` file in this repo. I suggest that you add one

Reply:
I was wondering what was that. I could have known.

Reply:
added 

I notice that in  https://github.com/akka/akka-persistence-r2dbc we have `.jvmopts`and is never used?